### PR TITLE
Add local storage for my portfolio

### PR DESCRIPTION
**Pull request by Besufekad-HAZ**
-Use the browser to store form data
-Use the local storage function to store my data
-Implement JSON.stringify to convert the value to a JSON string

### CODE REVIEW

File: file-storage.js [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+
+contactForm.addEventListener('input', () => {
+  const formData = {
+    firstName: fnameVal.value,
+    lastName: lnameVal.value,
+    email: emailVal.value,
+    message: msgVal.value
+  }
+  localStorage.setItem('userData', JSON.stringify(formData))
+})
```
Comment (Whoistolu) [inline file-storage.js:16]: - [ ] [REQUIRED] Kindly note that you are only setting the data/item to localStorage at the moment. A part of the project requirement also states that when the user loads the page, the code checks if there are any saved data in the local storage, and if so the input fields are pre-filled with the data. Kindly implement this with the window.onload() method. Below is a typical example:
```
window.onload = () => {
  let savedFormData = localStorage.getItem('contactFormInput');
  savedFormData = JSON.parse(savedFormData);
  // Check if the form data object is found on localStorage
  if (savedFormData) {
  // populate inputs values if data was found
  // ex: nameInput.value = savedFormData.name
  }
};
```

**Review by Whoistolu — CHANGES_REQUESTED** [review]
Hi Team,
@Besufekad-HAZ & @mj-e-boc 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️

- [ ]  _Check the inline comments under the review for sections to improve._

### Optional suggestions
Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better.

### N/A

Cheers and Happy coding!

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @whoistolu in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: file-storage.js (hunk 1 above)
Comment (Besufekad-HAZ) [reply file-storage.js:16]: Ok, thanks for the suggestion. I'll definitely address that.

File: file-storage.js [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+const object = localStorage.getItem('userData')
+const getValue = JSON.parse(object)
+
+window.addEventListener('load', () => {
+  if (localStorage.getItem('userData')) {
+    document.getElementById('f_name').value = getValue.firstName
+    document.getElementById('l_name').value = getValue.lastName
+    document.getElementById('mail').value = getValue.userEmail
```
Comment (huseyinbiyik) [inline file-storage.js]: Great job but let's fix the small email bug. It needs to be `email`, not 'userEmail` 😎

Screenshot 2023-02-02 at 11 06 23 PM

**Review by huseyinbiyik — CHANGES_REQUESTED** [review]
Hi team

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
Almost there

## Highlights
- Linter set up well and checks pass 🟢


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by brainconnect93 — APPROVED** [review]
## Status: Approved :heavy_check_mark:

Hi @Team, 👋 

Your project is complete! Great job for following the project requirement🥇. There is nothing else to say other than... it's time to merge it 🍾🚢 :shipit: . Congratulations!  🎉💯🌟 



## Good Points 👍

- Has a proper Pull Request. ✔️
- No Linter errors ✔️
- Good use of Github flow ✔️

### Changes Required ♻️

- [x]  _Please check the comments under the review._

### Optional Changes 🔥

Thank You.

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers..Happy coding!..and keep soaring higher! 💻 🍷 🚀 

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @brainconnect93 in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have 2 more limited reviews per this project. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**